src/ipcache.cc

```diff
@@ -445,7 +445,7 @@ ipcacheCallback(ipcache_entry *i, const bool hit, const int wait)
 
     if (hit)
         i->handler.forwardHits(i->addrs);
-    const Dns::LookupDetails details(i->error_message, wait);
+    const Dns::LookupDetails details(SBuf(i->error_message), wait);
```

Comment:
In order to use std::optional this caller code needs to determine whether an error message exists and construct LookupDetails with different constructors - one which sets error, and one which only sets wait time (not yet existing).

Note that this is now the third caller which needs to perform those same duplicate actions just to create a Dns::LookupDetails object "correctly" (avoiding setting the std::optional on non-errors). All of this extra complexity is imposed by using std::optional instead of the old API style of empty string check on use.